reaktive/src/commonMain/kotlin/com/badoo/reaktive/utils/queue/ArrayQueue.kt

```diff
@@ -1,16 +1,14 @@
 package com.badoo.reaktive.utils.queue
 
-import kotlin.math.abs
-
 internal class ArrayQueue<T> : Queue<T> {
 
-    private var queue: Array<T?> = createArray(8)
+    private var queue: Array<T?> = createArray(INITIAL_CAPACITY)
     private var head = 0
     private var tail = 0
     private var isFull = false
 
     val peek: T? get() = queue[head]
-    override val size: Int get() = abs(tail - head)
+    override val size: Int get() = if (isFull) queue.size else ((tail - head) and (queue.size - 1))
```

Comment:
What was the bug and how does this fix it? Would be useful if in issue or commit the bug was mentioned.

Reply:
Updated description of this issue, thanks

Reply:
Wouldn't it be easier to track the size in a variable? Is there an advantage to this approach?
Also a comment would be good for ((tail - head) and (queue.size - 1)). It's a pretty cool trick, took me a while.

Reply:
For me it looks more safe when you calculate instead of track. You can easily getting an inconsistency. I would like to even remove isFull flag, but it appears to be impossible.

Reply:
Fixed in a bit different way, please check